### PR TITLE
Fix sliceSepByMax

### PR DESCRIPTION
**Pull request by pranaysashank**
- Fix `sliceSepByMax`
- Add examples for `Parser.eqBy`

### CODE REVIEW

**Comment by pranaysashank** [comment]
Circle CI failure is due to the token change and can be ignored.

File: src/Streamly/Internal/Data/Parser.hs [hunk 1]
```diff
@@ -570,6 +570,11 @@ groupBy = undefined
 
 -- | Match the given sequence of elements using the given comparison function.
 --
+-- >>> S.parse $ S.eqBy (==) "string" $ S.fromList "string"
+--
+-- >>> S.parse $ S.eqBy (==) "mismatch" $ S.fromList "match"
+-- > *** Exception: ParseError "eqBy: failed, yet to match 6 elements"
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Parser.hs:576]: There is something wrong with the count? yet to match 6 elements?

File: src/Streamly/Internal/Data/Parser/ParserD.hs [hunk 2]
```diff
@@ -505,15 +505,17 @@ sliceSepByMax predicate cnt (Fold fstep finitial fextract) =
...
-            b <- fextract res
-            return $ Stop 0 b
+    step (Tuple' i r) a
+        | not (predicate a) = do
+            res <- fstep r a
+            let i1 = i + 1
+                s1 = Tuple' i1 res
+            if i1 < cnt
+                then return $ Yield 0 s1
+                else do
```
Comment (harendra-kumar) [inline src/Streamly/Internal/Data/Parser/ParserD.hs:515]: if-then-else style, don't indent `then` and `else`.

File: src/Streamly/Internal/Data/Parser.hs (hunk 1 above)
Comment (pranaysashank) [reply src/Streamly/Internal/Data/Parser.hs:576]: The word "match" isn't matched yet, and it has 6 letters in it.

**Pull request merged**